src/libmongoc/tests/test-mongoc-cursor.c

```diff
@@ -2350,6 +2350,119 @@ test_find_error_is_alive (void)
    mongoc_client_destroy (client);
 }
 
+typedef struct _started_event_t {
+   char *command_name;
+   bson_t *command;
+   struct _started_event_t *next;
+} started_event_t;
+
+static void
+command_started (const mongoc_apm_command_started_t *event)
+{
+   mongoc_array_t *events =
+      (mongoc_array_t *) mongoc_apm_command_started_get_context (event);
+   started_event_t *started_event = bson_malloc0 (sizeof (started_event_t));
+
+   started_event->command =
+      bson_copy (mongoc_apm_command_started_get_command (event));
+   started_event->command_name =
+      bson_strdup (mongoc_apm_command_started_get_command_name (event));
+   _mongoc_array_append_val (events, started_event);
+
+   MONGOC_DEBUG ("Command %s started",
+                 mongoc_apm_command_started_get_command_name (event));
+}
+
+static void
+clear_started_events (mongoc_array_t *events)
+{
+   for (size_t i = 0; i < events->len; i++) {
+      started_event_t *started_event =
+         _mongoc_array_index (events, started_event_t *, i);
+      bson_destroy (started_event->command);
+      bson_free (started_event->command_name);
+      bson_free (started_event);
+   }
+   _mongoc_array_clear (events);
+}
+
+/* Test that mongoc_cursor_set_batch_size overrides a previously set int32
+ * batchSize. */
+void
+test_cursor_batchsize_override_int32 (void)
+{
+   mongoc_client_t *client;
+   mongoc_apm_callbacks_t *cbs;
+   mongoc_collection_t *coll;
+   bson_error_t error;
+   mongoc_array_t started_events;
+
+   client = test_framework_new_default_client ();
+   cbs = mongoc_apm_callbacks_new ();
+   _mongoc_array_init (&started_events, sizeof (started_event_t *));
+   mongoc_apm_set_command_started_cb (cbs, command_started);
+   mongoc_client_set_apm_callbacks (client, cbs, &started_events);
+   coll = mongoc_client_get_collection (client, "db", "coll");
+
+   /* Drop and insert two documents into the collection */
+   {
+      bson_t *to_insert = BCON_NEW ("x", "y");
+
+      ASSERT_OR_PRINT (mongoc_collection_drop (coll, &error), error);
```

Comment:
```suggestion
      // Ignore "ns not found" error on drop.
      mongoc_collection_drop (coll, NULL);
```

Pre 7.0 MongoDB Servers may return "ns not found" error when attempting to drop a collection that does not exist. MongoDB 7.0 changed this behavior as part of DRIVERS-2548.

Suggest removing the assertion.